src/main/java/org/truffleruby/core/regexp/MatchDataNodes.java

```diff
@@ -167,17 +167,56 @@ protected Object create(Object regexp, Object string, int start, int end) {
     @Primitive(name = "matchdata_create")
     public abstract static class MatchDataCreateNode extends PrimitiveArrayArgumentsNode {
 
+        public static MatchDataCreateNode create() {
+            return MatchDataNodesFactory.MatchDataCreateNodeFactory.create(null);
+        }
+
+        public abstract Object executeMatchDataCreate(RubyRegexp regexp, Object string, Object starts, Object ends);
```

Comment:
Could `starts` and `ends` be in one array, allowing them to share storage guards?

Reply:
I think anyway we don't want to compute group offsets eagerly at all with TRegex (https://github.com/oracle/truffleruby/issues/2381#issuecomment-866843136), so this node should soon enough disappear.

Reply:
I removed those arrays altogether by inlining the MatchDataCreateNode and assigning to the Region directly.